docker/docker.go

```diff
@@ -302,3 +339,25 @@ func ReadContainerLogs(ctx context.Context, dClient *client.Client, containerNam
 	return dClient.ContainerLogs(ctx, containerName, types.ContainerLogsOptions{ShowStdout: true})
 
 }
+
+func tryRegistryAuth(pr v3.PrivateRegistry) types.RequestPrivilegeFunc {
+	return func() (string, error) {
+		return getRegistryAuth(pr)
+	}
+}
+
+func hasURL(imageName string) bool {
+	return strings.Contains(strings.Split(imageName, "/")[0], ".")
```

Comment:
there should be some standard library for parsing image, in docker, or k8s. Lets try to use it. Relying on prefix might not be a good idea.